tests/Spec.hs

```diff
@@ -1353,16 +1344,16 @@ main = hspec $ do
           state = ProjectState
             {
               Project.pullRequests         = prMap,
-              Project.integrationCandidate = Nothing,
               Project.pullRequestApprovalIndex = 2
             }
           -- Proceeding should pick the next pull request as candidate.
           results = defaultResults { resultIntegrate = [Right (Sha "38e")] }
           (state', actions) = runActionCustom results $ Logic.proceedUntilFixedPoint state
-          Just (cId, _candidate) = Project.getIntegrationCandidate state'
+          [(cId, _candidate)] = Project.getIntegrationCandidates state'
       cId     `shouldBe` PullRequestId 2
       actions `shouldBe`
-        [ ATryIntegrate "Merge #2: Add my test results\n\nApproved-by: deckard\nAuto-deploy: false\n" (Branch "refs/pull/2/head", Sha "f37") False
+        [ ATryPromote (Branch "results/leon") (Sha "38d")
```

Comment:
Where does this extra action come from?

Reply:
In this test suite, there are two open PRs:

* the first which has been rebased and the build has (just) passed;
* the second which was not rebased yet;

So the actions should be to promote the first to be the new master, and try to rebase (integrate) the second.

If we instead set the first PR to be `Promoted`, the `TryPromote` should disappear.

This extra action just came to how the state is represented now: before a `Promoted` pr was one that was `Integrated BuildSuceeded` while the candidate was `Nothing`.

I decided to change the test to include the promotion.

Reply:
I could update the test name, but I think it still applies :grin::

> "picks a new candidate from the queue after a successful push"